keps/sig-storage/1710-selinux-relabeling/README.md

```diff
@@ -818,26 +1054,166 @@ With a single `VolumeChangePolicy`, user has to fall back to `Recursive` policy
+            
+            # user: start with system_u if level is set
+            "final_selinux_user", "system_u", "final_selinux_level", "(.+)"),
+            # replace it by pod_selinux_user, if set
+            "final_selinux_user", "$1", "pod_selinux_user","(.+)"),
+            # replace it by container_selinux_user if it is set
+            "final_selinux_user", "$1", "container_selinux_user", "(.+)"),
+            # replace it by system_u for privileged pods
+            "final_selinux_user", "system_u", "privileged", "true"),
+            
+            # same for _role
+            "final_selinux_role", "system_r", "final_selinux_level", "(.+)"),
+            "final_selinux_role", "$1", "pod_selinux_role","(.+)"),
+            "final_selinux_role", "$1", "container_selinux_role", "(.+)"),
+            "final_selinux_role", "system_r", "privileged", "true"),
+            
+            # same for _type
+            "final_selinux_type", "container_t", "final_selinux_level", "(.+)"),
+            "final_selinux_type", "$1", "pod_selinux_type","(.+)"),
+            "final_selinux_type", "$1", "container_selinux_type", "(.+)"),
+            "final_selinux_type", "spc_t", "privileged", "true"),
+            
+            # This is label_join(): join all final_selinux_ labels into one string as final_selinux_label
+            "final_selinux_label", ":", "final_selinux_user", "final_selinux_role", "final_selinux_type", "final_selinux_level"),
+            # label_replace(): replace ":::" from the label_join above (= no label set in a pod) with "", just because it looks better
+            "final_selinux_label", "", "final_selinux_label", "^:::$")
+
+      # Add final_selinux_label to kube_pod_container_volume_mount.
+      # Augmented by csidriver.SELinuxMount, if set.
+      # This does a massive join from container -> pod -> pvc -> pv -> csidriver.
+      - record: kube_pod_container_volume_mount:selinux_label
+        expr: |
+          label_replace(
+            kube_pod_container_volume_mount
+              # Join final_selinux_label
+              * on(uid, pod, container) group_left (final_selinux_label) kube_pod_container_security_context:final_selinux_label
+              # Join CSIDriver.Spec.SELinuxMount by joining pvc -> pv -> csidriver
+              * on (uid, volume) group_left(persistentvolumeclaim) kube_pod_spec_volumes_persistentvolumeclaims_info
+              * on (namespace, persistentvolumeclaim) group_left(volumename) kube_persistentvolumeclaim_info
+              * on (volumename) group_left (csi_driver, csi_volume_handle, unique_volume_id) label_replace(kube_persistentvolume_info:unique_volume_id, "volumename", "$1", "persistentvolume", "(.+)")
+              * on(csi_driver) group_left (selinux_mount) kube_csidriver_info,
+                    
+            # label_replace: set final_selinux_label to "" when the CSI driver does not support SELinuxMount
+            "final_selinux_label", "", "selinux_mount", "false")
```

Comment:
This is a bit more tricky since this depends on both PV and pod collectors, and as such, two different core objects' attributes, defining metrics for which at the same place is not supported in KSM by design.